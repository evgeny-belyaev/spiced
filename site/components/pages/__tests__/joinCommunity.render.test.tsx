import renderer = require("react-test-renderer")
import React from "react"
import Page from "../../../../site/pages/join/[JoinConfirmationToken]"

export default describe("joinCommunity render test", () => {
    test("render", () => {
        // Arrange
        const props = {
            alreadyJoined: undefined,
            communityTitle: "communityTitle",
            error: undefined
        }

        // Act
        const tree = renderer.create(<Page {...props}/>).toJSON()

        // Assert
        expect(tree).toMatchSnapshot()
    })

    test("render alreadyJoined", () => {
        // Arrange
        const props = {
            alreadyJoined: true,
            communityTitle: "communityTitle",
            error: undefined
        }

        // Act
        const tree = renderer.create(<Page {...props}/>).toJSON()

        // Assert
        expect(tree).toMatchSnapshot()
    })

    test("render error", () => {
        // Arrange
        const props = {
            alreadyJoined: true,
            communityTitle: "communityTitle",
            error: "error"
        }

        // Act
        const tree = renderer.create(<Page {...props}/>).toJSON()

        // Assert
        expect(tree).toMatchSnapshot()
    })
})