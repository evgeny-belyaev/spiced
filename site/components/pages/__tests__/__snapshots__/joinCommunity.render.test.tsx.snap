// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joinCommunity render test render 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
    <table id=\\"allrecords\\" data-tilda-email=\\"yes\\" data-tilda-project-id=\\"3341502\\" data-tilda-page-id=\\"15719596\\"
       data-tilda-page-alias=\\"\\" cellpadding=\\"0\\" cellspacing=\\"0\\"
       style=\\"width:100%; border-collapse:collapse; border-spacing:0; padding:0; margin:0; border:0;\\">
    <tr>
        <td style=\\"background-color: #efefef; \\"><!--record_mail-->
            <table id=\\"rec257342934\\" style=\\"width:100%; border-collapse:collapse; border-spacing:0; margin:0; border:0;\\"
                   cellpadding=\\"0\\" cellspacing=\\"0\\" data-record-type=\\"619\\">
                <tr>
                    <td style=\\"padding-left:15px; padding-right:15px; \\">
                        <table id=\\"recin257342934\\" class=\\"r\\" style=\\"margin: 0 auto;border-spacing: 0;width:600px;\\"
                               align=\\"center\\">
                            <tr>
                                <td style=\\"padding-top:0px;padding-bottom:0px;padding-left:0;padding-right:0;\\">
                                    <table valign=\\"top\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\">
                                        <tr>
                                            <td style=\\"height:30px;\\" height=\\"30px\\"></td>
                                        </tr>
                                    </table>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
            </table><!--/record--><!--record_mail-->
            <table id=\\"rec257342935\\" style=\\"width:100%; border-collapse:collapse; border-spacing:0; margin:0; border:0;\\"
                   cellpadding=\\"0\\" cellspacing=\\"0\\" data-record-type=\\"622\\">
                <tr>
                    <td style=\\"padding-left:15px; padding-right:15px; background-color:#f7c304;\\">
                        <table id=\\"recin257342935\\" class=\\"r\\"
                               style=\\"margin: 0 auto;background-color:#f7c304;border-spacing: 0;width:600px;\\"
                               align=\\"center\\">
                            <tr>
                                <td style=\\"padding-top:45px;padding-bottom:45px;padding-left:0;padding-right:0;\\">
                                    <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\">
                                        <tr>
                                            <td class=\\"t-emailBlock\\" valign=\\"middle\\"
                                                style=\\"text-align: left; padding-right: 10px; width: 50%;\\"><a
                                                    style=\\"text-decoration: none; font-weight: normal; font-family: Helvetica Neue, Helvetica, Arial, sans-serif; color:#000000;font-size:30px;\\"
                                                    href=\\"https://tilda.cc/\\">tmixed</a></td>
                                            <td class=\\"t-emailBlock t-emailBlockPadding t-emailAlignLeft\\"
                                                valign=\\"middle\\"
                                                style=\\"text-align: right; padding-left: 10px; width: 50%;\\">
                                                <div style=\\" font-family: Helvetica Neue, Helvetica, Arial, sans-serif; color:#000000;font-size:16px;\\">
                                                    <strong>a brand new networking</strong></div>
                                            </td>
                                        </tr>
                                    </table>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
            </table><!--/record--><!--record_mail-->

            <table id=\\"rec256693177\\" style=\\"width:100%; border-collapse:collapse; border-spacing:0; margin:0; border:0;\\"
                   cellpadding=\\"0\\" cellspacing=\\"0\\" data-record-type=\\"323\\">
                <tr>
                    <td style=\\"padding-left:15px; padding-right:15px; \\">
                        <table id=\\"recin256693177\\" class=\\"r\\"
                               style=\\"margin: 0 auto;background-color:#ffffff;border-spacing: 0;width:600px;\\"
                               align=\\"center\\">
                            <tr>
                                <td style=\\"padding-top:30px;padding-bottom:15px;padding-left:30px;padding-right:30px;\\">
                                    <table valign=\\"top\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\"
                                           style=\\"width: 100%;\\">
                                        <tr>
                                            <td style=\\"text-align: left; padding: 0 0 0;\\">
                                                <div style=\\"margin-right: auto; font-family: Helvetica Neue, Helvetica, Arial, sans-serif; color:#222222;font-size:28px;font-weight:bold;\\">
                                                    You have successfully joined the communityTitle<span mc:edit=\\"communityTitle\\"></span>
                                                </div>
                                            </td>
                                        </tr>
                                    </table>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
            </table><!--/record--><!--record_mail-->
            <table id=\\"rec256693178\\" style=\\"width:100%; border-collapse:collapse; border-spacing:0; margin:0; border:0;\\"
                   cellpadding=\\"0\\" cellspacing=\\"0\\" data-record-type=\\"329\\">
                <tr>
                    <td style=\\"padding-left:15px; padding-right:15px; \\">
                        <table id=\\"recin256693178\\" class=\\"r\\"
                               style=\\"margin: 0 auto;background-color:#ffffff;border-spacing: 0;width:600px;\\"
                               align=\\"center\\">
                            <tr>
                                <td style=\\"padding-top:15px;padding-bottom:15px;padding-left:30px;padding-right:30px;\\">
                                    <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\" style=\\"width: 100%;\\">
                                        <tr>
                                            <td style=\\"text-align: left; padding: 0 0 0;\\">
                                                <div style=\\"margin-right: auto; font-family: Helvetica Neue, Helvetica, Arial, sans-serif; color:#444444;font-size:20px;line-height:1.45;\\">
                                                    That's all, thank you! <br/><br/> Get ready to meet your first
                                                    tmixed match! <br/>I am looking forward to sending it to you on
                                                    Monday.<br/></div>
                                            </td>
                                        </tr>
                                    </table>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
            </table><!--/record--><!--record_mail-->

            <table id=\\"rec257342938\\" style=\\"width:100%; border-collapse:collapse; border-spacing:0; margin:0; border:0;\\"
                   cellpadding=\\"0\\" cellspacing=\\"0\\" data-record-type=\\"627\\">
                <tr>
                    <td style=\\"padding-left:15px; padding-right:15px; \\">
                        <table id=\\"recin257342938\\" class=\\"r\\" style=\\"margin: 0 auto;border-spacing: 0;width:600px;\\"
                               align=\\"center\\">
                            <tr>
                                <td style=\\"padding-top:30px;padding-bottom:30px;padding-left:30px;padding-right:30px;\\">
                                    <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\">
                                        <tr>
                                            <td style=\\"text-align: center; padding-top: 25px;\\">
                                                <div style=\\"margin: 0 auto; font-weight: normal; font-family: Helvetica Neue, Helvetica, Arial, sans-serif; color:#a1a1a1;font-size:14px;\\">
                                                </div>
                                            </td>
                                        </tr>
                                    </table>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
            </table><!--/record--> </td>
    </tr>
</table><!--/allrecords-->
",
    }
  }
/>
`;

exports[`joinCommunity render test render alreadyJoined 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
    <table id=\\"allrecords\\" data-tilda-email=\\"yes\\" data-tilda-project-id=\\"3341502\\" data-tilda-page-id=\\"15719596\\"
       data-tilda-page-alias=\\"\\" cellpadding=\\"0\\" cellspacing=\\"0\\"
       style=\\"width:100%; border-collapse:collapse; border-spacing:0; padding:0; margin:0; border:0;\\">
    <tr>
        <td style=\\"background-color: #efefef; \\"><!--record_mail-->
            <table id=\\"rec257342934\\" style=\\"width:100%; border-collapse:collapse; border-spacing:0; margin:0; border:0;\\"
                   cellpadding=\\"0\\" cellspacing=\\"0\\" data-record-type=\\"619\\">
                <tr>
                    <td style=\\"padding-left:15px; padding-right:15px; \\">
                        <table id=\\"recin257342934\\" class=\\"r\\" style=\\"margin: 0 auto;border-spacing: 0;width:600px;\\"
                               align=\\"center\\">
                            <tr>
                                <td style=\\"padding-top:0px;padding-bottom:0px;padding-left:0;padding-right:0;\\">
                                    <table valign=\\"top\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\">
                                        <tr>
                                            <td style=\\"height:30px;\\" height=\\"30px\\"></td>
                                        </tr>
                                    </table>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
            </table><!--/record--><!--record_mail-->
            <table id=\\"rec257342935\\" style=\\"width:100%; border-collapse:collapse; border-spacing:0; margin:0; border:0;\\"
                   cellpadding=\\"0\\" cellspacing=\\"0\\" data-record-type=\\"622\\">
                <tr>
                    <td style=\\"padding-left:15px; padding-right:15px; background-color:#f7c304;\\">
                        <table id=\\"recin257342935\\" class=\\"r\\"
                               style=\\"margin: 0 auto;background-color:#f7c304;border-spacing: 0;width:600px;\\"
                               align=\\"center\\">
                            <tr>
                                <td style=\\"padding-top:45px;padding-bottom:45px;padding-left:0;padding-right:0;\\">
                                    <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\">
                                        <tr>
                                            <td class=\\"t-emailBlock\\" valign=\\"middle\\"
                                                style=\\"text-align: left; padding-right: 10px; width: 50%;\\"><a
                                                    style=\\"text-decoration: none; font-weight: normal; font-family: Helvetica Neue, Helvetica, Arial, sans-serif; color:#000000;font-size:30px;\\"
                                                    href=\\"https://tilda.cc/\\">tmixed</a></td>
                                            <td class=\\"t-emailBlock t-emailBlockPadding t-emailAlignLeft\\"
                                                valign=\\"middle\\"
                                                style=\\"text-align: right; padding-left: 10px; width: 50%;\\">
                                                <div style=\\" font-family: Helvetica Neue, Helvetica, Arial, sans-serif; color:#000000;font-size:16px;\\">
                                                    <strong>a brand new networking</strong></div>
                                            </td>
                                        </tr>
                                    </table>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
            </table><!--/record--><!--record_mail-->

            <table id=\\"rec257431099\\" style=\\"width:100%; border-collapse:collapse; border-spacing:0; margin:0; border:0;\\"
                   cellpadding=\\"0\\" cellspacing=\\"0\\" data-record-type=\\"323\\">
                <tr>
                    <td style=\\"padding-left:15px; padding-right:15px; \\">
                        <table id=\\"recin257431099\\" class=\\"r\\"
                               style=\\"margin: 0 auto;background-color:#ffffff;border-spacing: 0;width:600px;\\"
                               align=\\"center\\">
                            <tr>
                                <td style=\\"padding-top:30px;padding-bottom:15px;padding-left:30px;padding-right:30px;\\">
                                    <table valign=\\"top\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\"
                                           style=\\"width: 100%;\\">
                                        <tr>
                                            <td style=\\"text-align: left; padding: 0 0 0;\\">
                                                <div style=\\"margin-right: auto; font-family: Helvetica Neue, Helvetica, Arial, sans-serif; color:#222222;font-size:28px;font-weight:bold;\\">
                                                    Hey, you are not alone!<br/>You have already joined the communityTitle
                                                </div>
                                            </td>
                                        </tr>
                                    </table>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
            </table><!--/record--><!--record_mail-->

            <table id=\\"rec257342938\\" style=\\"width:100%; border-collapse:collapse; border-spacing:0; margin:0; border:0;\\"
                   cellpadding=\\"0\\" cellspacing=\\"0\\" data-record-type=\\"627\\">
                <tr>
                    <td style=\\"padding-left:15px; padding-right:15px; \\">
                        <table id=\\"recin257342938\\" class=\\"r\\" style=\\"margin: 0 auto;border-spacing: 0;width:600px;\\"
                               align=\\"center\\">
                            <tr>
                                <td style=\\"padding-top:30px;padding-bottom:30px;padding-left:30px;padding-right:30px;\\">
                                    <table border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" width=\\"100%\\">
                                        <tr>
                                            <td style=\\"text-align: center; padding-top: 25px;\\">
                                                <div style=\\"margin: 0 auto; font-weight: normal; font-family: Helvetica Neue, Helvetica, Arial, sans-serif; color:#a1a1a1;font-size:14px;\\">
                                                </div>
                                            </td>
                                        </tr>
                                    </table>
                                </td>
                            </tr>
                        </table>
                    </td>
                </tr>
            </table><!--/record--> </td>
    </tr>
</table><!--/allrecords-->
",
    }
  }
/>
`;

exports[`joinCommunity render test render error 1`] = `"Server error"`;
